### ISSUE TITLE
terrainSource maxZoom 

### ISSUE THREAD

**Issue by bolollo** [ted-piotrowski/mapbox-gl-shadow-simulator#1]
Hi, I am doing some tests with MapLibre and MapTiler's terrain, and the maxZoom value in terrainSource does not respect it. I have put 12 and he makes requests to me up to level 14.

```
terrainSource: {
    tileSize: 256,
    maxZoom: 12,
    getSourceUrl: ({x, y, z}) => {
      return `https://api.maptiler.com/tiles/terrain-rgb-v2/${z}/${x}/${y}.webp?key=${key}`;
    },
    getElevation: ({r, g, b, a}) => {
      return -10000 + ((r * 256 * 256 + g * 256 + b) * .1);
    }
  },
```

Another question, what license does the library have?

**Comment by ted-piotrowski** [comment]
Could you change maxZoom to 10, does it still zoom to 14 or does it stop at another zoom level like 12 now?

The code is unlicensed. It's free for educational and personal projects and I charge a licensing fee for commercial use. You can read more here: https://shademap.app/about/#pricing Reach out if you have any questions.

I took a quick glance at the code and I don't see an obvious bug, but I will update once I can reproduce. Thanks for taking the time to report the issue.

**Comment by bolollo** [comment]
I've tried setting the maxZoom to 10, and it keeps asking until 14. I've also tried it with the mapbox-gl library and it does the same thing.

I wanted to make an example with maplibre-gl-js and MapTiler Terrain and add it to the readme. Do you agree?

**Comment by ted-at-felt** [comment]
I managed to track down the bug and released v0.13.5 with the fix. I also took the liberty of updating the README.md to include Maptiler configuration per your suggestion.

**Comment by ted-piotrowski** [comment]
I managed to track down the bug and released a fix in v0.13.5. I also took the liberty of updating the README with a Maptiler configuration https://github.com/ted-piotrowski/mapbox-gl-shadow-simulator#maptiler-terrain-rgb-v2

Let me know if you're still having issues and thanks a lot for reporting this issue.

**Comment by bolollo** [comment]
I have tested it, and it works fine :+1:. Thanks for adding the MapTiler example to the README.
Thank you very much, and I hope it goes well with the project.
